Makefile

```diff
@@ -301,6 +304,16 @@ lint-license: pull-buildenv buildenv-dirs
 .PHONY: install-kustomize
 install-kustomize: "$(BIN_DIR)/kustomize"
 
+"$(GOBIN)/kind":
+	go install sigs.k8s.io/kind@$(KIND_VERSION)
+
+# install kind binary for containerized testing
+"$(BIN_DIR)/kind": "$(GOBIN)/kind" buildenv-dirs
+	cp $(GOBIN)/kind $(BIN_DIR)/kind
+
+.PHONY: install-kind
+install-kind: "$(BIN_DIR)/kind"
```

Comment:
this target seems like it might be confusing if used locally. how about just specifying one of the above?

Reply:
Alternatively: make install-kind (GOBIN) & install-kind-ci (BIN_DIR)

Reply:
This follows the pattern of other similar make targets that install local binaries. The `install-kind` target is a phony target to make it easier to call from the command line. The other targets aren't phony and point to the actual path of the binary

Reply:
I understand. My complaint is that this install target is a phony target for use by CI, not developers locally. 

If we only have one, I think it should detect CI or not CI. Otherwise, have two, one with a -ci suffix.

Reply:
That said, I see kustomize has the same problem...

Reply:
The intent of this make target is to also be used by developers locally

Reply:
The reason it gets installed both in `GOBIN` and `BIN_DIR` is for legacy reasons at this point, not necessarily for local vs CI use case. Some of the make targets for running tests used to mount `BIN_DIR` as a path when running the targets inside a docker container - this would be the same locally and in CI.

Reply:
I updated this to add a separate CI target, although I don't really feel like this is necessary. The same `install-kind` target should be usable in both CI and local use cases.

I suppose with how finicky `kind` can be with changing versions, it's okay to force the developer to separately call `install-kind`. I was thinking it would be a convenience to have the testing targets ensure `kind` is installed before running the tests

Reply:
I think the other make targets all use `"$(BIN_DIR)/kustomize"` when executing kustomize commands, which uses what's installed in the the output dir, regardless of what's in your PATH.

But for kind, I think they all just call kind, which uses what's in your PATH, regardless of what's in your output dir.

For kustomize, I almost never need to run it manually. I just use the make targets to build the manifests.

But for kind, I use the kind command in my path all the time, to list clusters, delete clusters, and export the kubeconfig. So I'd much rather use the kind in my PATH, rather than have to specify the exact path to the kind binary in the output dir...

So it only makes sense to treat kind and kustomize the same if I never have to call kind directly, which isn't the case today.

Reply:
All of the make target entrypoints will use the `kind` binary in BIN_DIR, because it's [injected in the PATH](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/Makefile#L175). 

Why do you need to specify the exact for the kind binary in the output dir? For local command line I do what's described in the [dev docs](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/docs/development.md#e2e-tests-kind), and put `GOPATH/bin` in my `PATH`. `install-kind` puts the kind binary in both `GOPATH/bin` and `.output/go/bin`

Reply:
Ah, I see it's installing to GOBIN and then copying to BIN_DIR. 

I assumed it was installing to BIN_DIR using a modified GOPATH, which is how it originally worked, IIRC.

It looks like it's all be changed and my understanding of how it works is out of date.

So it's probably fine. Sorry for the confusion.

Reply:
You can move install-kind back to be a dependency of config-sync-manifest-local.
If it's installing in GOBIN and copying to BIN_DIR it should work fine locally and in CI.

Reply:
No worries, moved it back to `config-sync-manifest-local`